docs/01-app/03-api-reference/03-file-conventions/loading.mdx

```diff
@@ -85,14 +85,57 @@ In the same folder, `loading.js` will be nested inside `layout.js`. It will auto
+- Otherwise, [streaming metadata](/docs/app/api-reference/functions/generate-metadata#streaming-metadata) may be used used. Next.js automatically detects user agents to choose between blocking and streaming behavior.
 - Since streaming is server-rendered, it does not impact SEO. You can use the [Rich Results Test](https://search.google.com/test/rich-results) tool from Google to see how your page appears to Google's web crawlers and view the serialized HTML ([source](https://web.dev/rendering-on-the-web/#seo-considerations)).
 
 ### Status Codes
 
 When streaming, a `200` status code will be returned to signal that the request was successful.
 
-The server can still communicate errors or issues to the client within the streamed content itself, for example, when using [`redirect`](/docs/app/api-reference/functions/redirect) or [`notFound`](/docs/app/api-reference/functions/not-found). Since the response headers have already been sent to the client, the status code of the response cannot be updated. This does not affect SEO.
+The server can still communicate errors or issues to the client within the streamed content itself, for example, when using [`redirect`](/docs/app/api-reference/functions/redirect) or [`notFound`](/docs/app/api-reference/functions/not-found). Because the response headers have already been sent to the client, the status code of the response cannot be updated.
+
+When a route resolves to a missing resource using [`notFound`](/docs/app/api-reference/functions/not-found), Next.js includes a `<meta name="robots" content="noindex">` tag in the streamed HTML. This prevents search engines from indexing that URL even if the HTTP status is 200. See Google’s guidance on the [`robots` meta tag](https://developers.google.com/search/docs/crawling-indexing/robots-meta-tag).
+
+Some crawlers may label these responses as “soft 404s”. In the streaming case, this does not lead to indexation because the page is explicitly marked `noindex` in the HTML.
+
+If you need a 404 status, for compliance or analytics, ensure the resource exists before the response body is streamed, so that the server can set the HTTP status code.
+
+<details>
+<summary>When is the response body streamed?</summary>
+
+The response body starts streaming when a Suspense fallback renders (for example, a `loading.tsx`) or when a Server Component suspends under a Suspense boundary. Place `notFound()` before those boundaries and before any `await` that may suspend.
+
+To start streaming, the response headers must be set. This is why it is not possible to change the status code after streaming started.
+
+</details>
+
+Place a [`notFound`](/docs/app/api-reference/functions/not-found) guard as close to the top as possible in both [`generateMetadata`](/docs/app/api-reference/functions/generate-metadata) and your page or layout component, and avoid being inside any Suspense boundary, for example a `loading.tsx`, when doing this check. This ensures a 404 can be set before streaming:
```

Comment:
It's worth mentioning that this is not a call to action. Someone might decide that this is the best practice